Cargo.toml

```diff
@@ -29,14 +29,20 @@ config = "0.13.3"
 env_logger = "0.10.0"
 hex = "0.4.3"
 jwt-simple = "0.11"
-kbs_protocol = { git = "https://github.com/confidential-containers/guest-components.git", rev="e6999a3c0fd877dae9e68ea78b8b483062db32b8", default-features = false }
+kbs_protocol = { git = "https://github.com/Xynnn007/guest-components.git", rev = "2bfe3559bf273ddc835817498843f49c85359bbf", default-features = false }
```

Comment:
I originally envisioned the opposite order because with Trustee changes available first, GC can test against `:latest` image that has the necessary changes available.

Reply:
Oh, ok I misunderstood your point. As now we imports GC code to cover e2e test of kbs. We could only choose the GC reference to repo rather than CoCo's for this PR to make the e2e test pass. Or, it will refer to the old version of GC code thus failed like [this](https://github.com/confidential-containers/trustee/actions/runs/12002528508/job/33454729423) due to different decryption logic of JWE.

So there are two ways
1. merge KBS side code firstly with gc git reference of personal development branch. Then merge GC side. Then go back to KBS side to change the gc reference to CoCo's
2. merge GC side code firstly with kbs image built from personal development code of kbs. Then merge kbs side. Then go back to gc to change back the image reference.


I think either is good to me.


What is not pretty clear to me are
1. Do we need to update minor number of kbs protocol for this patch? I guess yes. But the tricky thing is that the document of protocol is not bad, but the code.
2. Do we need to make a change log file for historical kbs protocol updates?

Reply:
> 1. Do we need to update minor number of kbs protocol for this patch? I guess yes. But the tricky thing is that the document of protocol is not bad, but the code.

I think yes, since this is pretty close to a schema change since tag was always an empty string. 

> 2. Do we need to make a change log file for historical kbs protocol updates?

Maybe it's sufficient to have a good commit message that will then end up in a generated changelog

> Do we want to move kbs_protocol crate to trustee, or combine GC and trustee repos in some future?

If we move the kbs_protocol crate to this repo, we wouldn't have a dependency from trustee to guest-components anymore? If that's the case, I think that would be a way to break the circular dependency, if I don't miss anything.

then let's maybe move the crate (in a separate PR) before we merge this PR?

Reply:
> I originally envisioned the opposite order because with Trustee changes available first, GC can test against `kbs:latest` image that has the necessary changes available.

Actually, sorry. I checked the `kbs-types` bump case and it was indeed like you suggested: GC was merged first.

This is not only about keeping CI happy but also to notify users "both sides need updating" (which is important now as Trustee and GC can have their own releases). Without a `ProtocolVersion` error, users might just see an error without a clue what's wrong. Only `patch` version bump would be sufficient to make progress.

Reply:
> If we move the kbs_protocol crate to this repo, we wouldn't have a dependency from trustee to guest-components anymore? If that's the case, I think that would be a way to break the circular dependency, if I don't miss anything.

We still needs to import `attester`, `resource_uri` and `crypto` from GC. Whether these dependencies will cause problems is still open to question. If we need a quick fix on `attester` to make AA work, it would make us to have two PRs -- one in GC and one in trustee.

Reply:
right, I think that's too much of a refactoring then. I suspect with the kbs-client moving into dedicated set of tools (potentially into another repo), it's maybe better to wait how this will look like.